### PR TITLE
Added Load test for PubsubToAvro template

### PR DESCRIPTION
**Pull request by Amar3tto**
- Implemented `PubsubToAvroLT` test

Do not merge until `gs://dataflow-templates/latest/flex/Cloud_PubSub_to_Avro_Flex` is available

### CODE REVIEW

File: it/google-cloud-platform/src/main/java/com/google/cloud/teleport/it/gcp/datagenerator/DataGenerator.java
Comment (Lizzfox) [inline it/google-cloud-platform/src/main/java/com/google/cloud/teleport/it/gcp/datagenerator/DataGenerator.java:105]: could you please add a justification for this update in the PR description?
something like this: the method can throw an NPE exception. Added try-catch statement

File: v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java
Comment (Lizzfox) [inline v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java:72]: `TIMEOUT_MIN_FOR_10_GB_TEST` - why "MIN"? Let's leave just 'TIMEOUT_FOR_10_GB_TEST'

File: v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java
Comment (Lizzfox) [inline v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java:147]:
```suggestion
    // Wait for the template to start
```

File: v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java
Comment (Amar3tto) [reply v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java:72]: I mean in minutes

File: v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java
Comment (pranavbhandari24) [reply v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java:151]: Just realized that moving the datagenerator before running the pipeline would result in no messages in the Pub/Sub to Avro pipeline (since the pipeline does not support passing in a subscription due to https://github.com/GoogleCloudPlatform/DataflowTemplates/issues/723, and if a topic is provided it creates a new subscription)

I'm working on solving that issue

**Comment by Abacn** [comment]
Let me add a Google LGTM tag to make the PR imported for internal tests

**Comment by bvolpato** [comment]
Thanks! Can we please squash the commits into a single one? Our current tooling doesn't allow squash on merging, so the history of commits/fixups would be imported to our main history, which is often not desired.

image

File: v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java
Comment (pranavbhandari24) [reply v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java:145]: Let's change the current test and update the spec path to `gs://dataflow-templates-southamerica-west1/latest/flex/Cloud_PubSub_to_Avro_Flex`

Let's also move this test to `v2/googlecloud-to-googlecloud`,

File: v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java
Comment (pranavbhandari24) [inline v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java:144]: This sleep is not required, we can remove this.

File: v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java
Comment (pranavbhandari24) [inline v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java:197]: Same here

**Review by pranavbhandari24 — CHANGES_REQUESTED** [review]
Looks good! Just needs a couple changes,

1. Let's change the spec path to point to the flex template.
2. Let's move the test file to v2/googlecloud-to-googlecloud as the flex template exists there.
3. Move the `datagenerator.execute(..)` to before launching the pipeline under test in `testBacklog10gb`
4. Remove the sleep for workers in both the tests.

File: v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java
Comment (Amar3tto) [reply v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java]: Done

File: v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java
Comment (Amar3tto) [reply v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java]: Done

File: v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java
Comment (Amar3tto) [reply v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java]: Done

**Review by pranavbhandari24 — APPROVED** [review]
Looks good! Thanks!
Let's wait for the current release is complete before merging!

**Review by Precha24 — APPROVED** [review]
Good, thanks

**Pull request merged**
